src/common/components/repositories-home/index.js

```diff
@@ -28,7 +24,7 @@ class RepositoriesHome extends Component {
     });
   }
 
-  componentDidMount() {
+  componentWillMount() {
```

Comment:
We used to do initial data fetch in `componentDidMount`, but it resulted in rendering 'empty' state without the spinner because data was not loading yet.

Moving it to `componentWillMount` triggers actions earlier so that redux store updates before we do initial render. Even though data is not fetched yet, we at least know that fetching started and component renders spinner properly.